components/SelectField.tsx

```diff
@@ -0,0 +1,150 @@
+import { FC, useState, useEffect, ReactNode } from 'react'
+
+import {
+  Box,
+  Flex,
+  FlexProps,
+  HTMLChakraProps,
+  Popover,
+  PopoverBody,
+  PopoverContent,
+  PopoverTrigger,
+  useDisclosure,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+import { TriangleDownIcon } from '@chakra-ui/icons'
+
+type OptionType = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  value: any
+  label: string | ReactNode
+  helperText: string | ReactNode
+}
+
+interface OptionProps extends HTMLChakraProps<'div'> {
```

Comment:
For my own edification, this extension is to basically allow for a ChakraComponent with an `option` prop?

Reply:
Yes.